components/activities/activities-tab/activities-tab.tsx

```diff
@@ -0,0 +1,112 @@
+import { React, useSelector, useStyleSheet } from "../../../deps.ts";
+import { usePollingActivityAlertConnection$ } from "../signals.ts";
+import { isActiveActivity, isPrediction } from "../../../shared/mod.ts";
+
+import PollListItem from "../poll-list-item/poll-list-item.tsx";
+import RaidListItem from "../raid-list-item/raid-list-item.tsx";
+import styleSheet from "./activities-tab.scss.js";
+import { StringKeys } from "../../../types/mod.ts";
+
+export interface ActivityListItemProps<ActivityType> {
+  activity: ActivityType;
+  isActive?: boolean;
+}
+
+type ListItemMap<ActivityTypes> = {
+  [K in keyof ActivityTypes]: (props: ActivityListItemProps<ActivityTypes[K]>) => JSX.Element;
+};
+
+export const DEFAULT_LIST_ITEMS = {
+  poll: PollListItem,
+  alert: RaidListItem,
+};
+
+export default function ActivitiesTab() {
+  return <ActivitiesTabManager activityListItems={DEFAULT_LIST_ITEMS} />;
+}
+
+interface ActivitiesTabManagerProps<ActivityTypes> {
+  activityListItems: ListItemMap<ActivityTypes>;
+}
+export function ActivitiesTabManager<
+  ActivityListItemTypes,
+  SourceType extends StringKeys<ActivityListItemTypes>,
+  ActivityType extends ActivityListItemTypes[SourceType],
+>(props: ActivitiesTabManagerProps<ActivityListItemTypes>) {
+  useStyleSheet(styleSheet);
+  const activityListItems = props.activityListItems ?? DEFAULT_LIST_ITEMS;
+
+  const activityAlertConnection$ = usePollingActivityAlertConnection$<ActivityType, SourceType>(
+    2000,
+    20,
+  );
+
+  const activeActivities = useSelector(() =>
+    activityAlertConnection$.data?.get()?.alertConnection?.nodes?.filter((activity) =>
```

Comment:
i feel like w/ graphql and typescript we should be able to know if any of these will be undef and have less optional chaining